app/qml/map/MapWrapper.qml

```diff
@@ -710,6 +710,13 @@ Item {
           {
             // just banner
             syncSuccessfulBanner.show()
+
+            // check for confliected copies and reload project if any to
+            // reset connections to the GPKG files (see #1798)
+            if ( __activeProject.hasConflictedCopies() )
+            {
+              __activeProject.reloadProject( __activeProject.qgsProject.homePath )
+            }
```

Comment:
It would be good to have this kind of logic in c++ rather than in qml - maybe inside ActiveProject class?

Reply:
Moved to C++.